network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt

```diff
@@ -1,15 +1,50 @@
 package com.wire.kalium.network
 
+import com.wire.kalium.network.api.base.model.ProxyCredentialsDTO
+import com.wire.kalium.network.tools.ServerConfigDTO
 import io.ktor.client.engine.HttpClientEngine
 import io.ktor.client.engine.okhttp.OkHttp
 import okhttp3.OkHttpClient
+import java.net.Authenticator
+import java.net.InetSocketAddress
+import java.net.PasswordAuthentication
+import java.net.Proxy
 import java.util.concurrent.TimeUnit
 
-actual fun defaultHttpEngine(): HttpClientEngine = OkHttp.create {
+actual fun defaultHttpEngine(
+    serverConfigDTOLinks: ServerConfigDTO.Links?,
```

Comment:
Why pass in `ServerConfigDTO.Links` when you always access `ServerConfigDTO.Links.Proxy`?